ADAL/src/cache/ADTokenCache.m

```diff
@@ -567,7 +603,7 @@ - (BOOL)addOrUpdateImpl:(ADTokenCacheItem *)item
 
 - (NSString *)description
 {
-    return [NSString stringWithFormat:@"ADTokenCache: %@", _cache];
+    return [NSString stringWithFormat:@"ADTokenCache: %@", [self cacheCopy]];
```

Comment:
As a general rule making extra copies of things should be avoided unless making that copy is going to be faster then what you're trying to do. In -serialize that's probably the case, most everywhere else probably not.